pip/compat.py

```diff
@@ -41,7 +46,11 @@ def console_to_str(s):
         try:
             return s.decode(sys.__stdout__.encoding)
         except UnicodeDecodeError:
-            return s.decode('utf_8')
+            if WINDOWS:
+                from ctypes import cdll
+                return s.decode("cp" + str(cdll.kernel32.GetACP()), 'replace')
```

Comment:
Well, that's for Python programs, but what do other programs (like the C compiler) use? I thought that console programs were supposed to use the OEM code page.

I guess it doesn't matter that much either way, though. If we guess wrong (and it's never better than a guess) then the worst that happens is that the output is a bit garbled. But that's still better than what currently happens.